cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -507,6 +507,37 @@ impl MachInstEmit for Inst {
                     sink,
                 );
             }
+            &Inst::Shl64 { rd, rs1, rs2 } => {
+                let rs1 = allocs.next(rs1);
+                let rs2 = allocs.next(rs2);
+                debug_assert_eq!(rs1, a0());
+                debug_assert_eq!(rs2, e0());
+                let rd = allocs.next_writable(rd);
```

Comment:
Why is this code not using `rd`? In my understanding, this is where Cranelift expects the result to be written.

Reply:
Fixed ``rd`` was used because it is register which is returned from last multiplication, and this will decrease (at least not increase) amount of moves (``<reg1> => <reg2>``)

Reply:
I mean, if ``ARITH`` leave answer in register ``X``, I can make ``X => {op}``, but cranelift will know nothing about it, and maybe assign ``op`` as ``Y``, and in next instruction lowering again move ``Y => X``. I think it's better do not do extra manual moves and just allow cranelift generate all moves by itself

Reply:
I tried, but didn't find how to cast ``Writable<Reg>`` to ``Reg`` (=